packages/flutter/test/material/text_button_test.dart

```diff
@@ -1009,6 +1009,46 @@ void main() {
     expect(paddingWidget.padding, const EdgeInsets.all(22));
   });
 
+  testWidgets('TextButton.icon is configurable when iconAlignment set to ButtonIconAlignment.start', (WidgetTester tester) async {
+    const IconData iconData = Icons.add;
+    const ButtonIconAlignment buttonIconAlignment = ButtonIconAlignment.start;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Center(
+          child: TextButton.icon(
+            onPressed: () { },
+            icon: const Icon(iconData),
+            iconAlignment: buttonIconAlignment,
+            label: const Text('text button'),
+          ),
+        ),
+      ),
+    );
+    final dynamic textButtonWithIconWidget = tester.widget(find.byWidgetPredicate((Widget widget) => '${widget.runtimeType}' == '_TextButtonWithIconChild'));
+    expect(textButtonWithIconWidget.iconAlignment, buttonIconAlignment);
+  });
+
+  testWidgets('TextButton.icon is configurable when iconAlignment set to ButtonIconAlignment.end', (WidgetTester tester) async {
+    const IconData iconData = Icons.add;
+    const ButtonIconAlignment buttonIconAlignment = ButtonIconAlignment.end;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Center(
+          child: TextButton.icon(
+            onPressed: () { },
+            icon: const Icon(iconData),
+            iconAlignment: buttonIconAlignment,
+            label: const Text('text button'),
+          ),
+        ),
+      ),
+    );
+    final dynamic textButtonWithIconWidget = tester.widget(find.byWidgetPredicate((Widget widget) => '${widget.runtimeType}' == '_TextButtonWithIconChild'));
+    expect(textButtonWithIconWidget.iconAlignment, buttonIconAlignment);
+  });
+
 }
 
```

Comment:
your tests verify that the value is passed into the _TextButtonWithIconChild, but not that the _TextButtonWithIconChild itself is doing anything with it, which is really the most important thing to check. :-)

Reply:
Ah okay I get it, I will try to check that the items place changes:
* when `iconAlignment` is set to `ButtonIconAlignment.start` it should be [Icon, SizedBox, Text].
* when `iconAlignment` is set to `ButtonIconAlignment.end` it should be [Text, SizedBox, Icon].
I think this will be good, will add it and let you know 🚀.

Reply:
Great news 🎉. I finally got some time to do the test and I think they work great! I hope you can take a look.